src/repr/src/adt/numeric.rs

```diff
@@ -762,11 +763,76 @@ impl DecimalLike for Numeric {
     }
 }
 
+/// An encoded packed variant of [`Numeric`].
+///
+/// Unlike other "Packed" types we _DO NOT_ uphold the invariant that
+/// [`PackedNumeric`] sorts the same as [`Numeric`].
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+pub struct PackedNumeric(pub [u8; 40]);
+
+impl FixedSizeCodec<Numeric> for PackedNumeric {
+    const SIZE: usize = 40;
+
+    fn as_bytes(&self) -> &[u8] {
+        &self.0
+    }
+
+    fn from_bytes(slice: &[u8]) -> Result<Self, String> {
+        let buf: [u8; Self::SIZE] = slice.try_into().map_err(|_| {
+            format!(
+                "size for PackedNumeric is {} bytes, got {}",
+                Self::SIZE,
+                slice.len()
+            )
+        })?;
+        Ok(PackedNumeric(buf))
+    }
+
+    fn from_value(val: Numeric) -> PackedNumeric {
+        let (digits, exponent, bits, lsu) = val.to_raw_parts();
+
+        let mut buf = [0u8; 40];
+
+        buf[0..4].copy_from_slice(&digits.to_le_bytes());
+        buf[4..8].copy_from_slice(&exponent.to_le_bytes());
+
+        for i in 0..13 {
+            buf[(i * 2) + 8..(i * 2) + 10].copy_from_slice(&lsu[i].to_le_bytes());
+        }
+
+        buf[34..35].copy_from_slice(&bits.to_le_bytes());
+
+        PackedNumeric(buf)
+    }
+
+    fn into_value(self) -> Numeric {
+        let digits: [u8; 4] = self.0[0..4].try_into().unwrap();
+        let digits = u32::from_le_bytes(digits);
+
+        let exponent: [u8; 4] = self.0[4..8].try_into().unwrap();
+        let exponent = i32::from_le_bytes(exponent);
+
+        let mut lsu = [0u16; 13];
+        for i in 0..13 {
+            let x: [u8; 2] = self.0[(i * 2) + 8..(i * 2) + 10].try_into().unwrap();
+            let x = u16::from_le_bytes(x);
+            lsu[i] = x;
+        }
+
+        let bits: [u8; 1] = self.0[34..35].try_into().unwrap();
+        let bits = u8::from_le_bytes(bits);
+
+        Numeric::from_raw_parts(digits, exponent, bits, lsu)
+    }
```

Comment:
Only tested in unit tests and looks very complex! Might be a good candidate for SLT/TD

Reply:
Only being covered in unit tests is expected! This encoding will be used for the V1 of columnar encodings which isn't fully hooked up yet